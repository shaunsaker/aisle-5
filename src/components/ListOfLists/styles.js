import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  contentContainer: {},
  headerContainer: {
    justifyContent: 'center',
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
  },
  itemContainer: {
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
  },
});

export default styles;
