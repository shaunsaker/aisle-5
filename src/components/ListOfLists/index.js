import React from 'react';
import PropTypes from 'prop-types';
import { SectionList, View } from 'react-native';

import styles from './styles';

import LinkText from '../LinkText';
import ShoppingItem from '../ShoppingItem';
import ItemSeparator from '../ItemSeparator';

export default class ListOfLists extends React.Component {
  constructor(props) {
    super(props);

    this.renderSectionHeader = this.renderSectionHeader.bind(this);
    this.renderItem = this.renderItem.bind(this);
    this.renderItemSeparator = this.renderItemSeparator.bind(this);

    this.itemHeight = 51;

    this.state = {};
  }

  static propTypes = {
    sections: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string,
        data: PropTypes.arrayOf(PropTypes.shape({})),
      }),
    ).isRequired,
  };

  static defaultProps = {};

  renderSectionHeader({ section }) {
    return (
      <View style={styles.headerWrapper}>
        <View style={[styles.headerContainer, { height: this.itemHeight }]}>
          <LinkText text={section.title} disabled />
        </View>

        <ItemSeparator />
      </View>
    );
  }

  renderItem({ item }) {
    return (
      <View style={styles.itemWrapper}>
        <View style={styles.itemContainer}>
          <ShoppingItem name={item.name} quantity={item.quantity} height={this.itemHeight} />
        </View>

        <ItemSeparator />
      </View>
    );
  }

  renderItemSeparator() {
    return <ItemSeparator />;
  }

  render() {
    const { sections } = this.props;

    return (
      <SectionList
        ref={(c) => {
          this.flatList = c;
        }}
        keyExtractor={({ id }) => id}
        sections={sections}
        renderSectionHeader={this.renderSectionHeader}
        renderItem={this.renderItem}
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
        bounces={false}
        stickySectionHeadersEnabled
      />
    );
  }
}
