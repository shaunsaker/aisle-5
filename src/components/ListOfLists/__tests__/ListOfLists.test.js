import React from 'react';
import renderer from 'react-test-renderer';

import ListOfLists from '..';
import ITEM from '../../../mockData/item';

describe('ListOfLists', () => {
  const sections = [
    {
      title: 'Test',
      data: [ITEM],
    },
  ];

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<ListOfLists sections={sections} />);

      expect(component).toMatchSnapshot();
    });
  });
});
