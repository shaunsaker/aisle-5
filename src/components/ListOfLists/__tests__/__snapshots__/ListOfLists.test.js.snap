// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfLists renders renders with minimum required props 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "id": "1",
            "isChecked": true,
            "name": "Beer",
            "quantity": 6,
          },
        ],
        "title": "Test",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  renderSectionHeader={[Function]}
  scrollEventThrottle={50}
  sections={
    Array [
      Object {
        "data": Array [
          Object {
            "id": "1",
            "isChecked": true,
            "name": "Beer",
            "quantity": 6,
          },
        ],
        "title": "Test",
      },
    ]
  }
  stickyHeaderIndices={Array []}
  stickySectionHeadersEnabled={false}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#EDE8CA",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "justifyContent": "center",
                "paddingHorizontal": 15,
              },
              Object {
                "height": 51,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#08293A",
                    "fontFamily": "Muli-Bold",
                    "fontSize": 15,
                    "textDecorationLine": "underline",
                  },
                  undefined,
                ]
              }
            >
              TEST
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#FFFFFF",
              "height": 1,
            }
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 15,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "height": 51,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#212121",
                  "fontFamily": "Muli-Bold",
                  "fontSize": 17,
                }
              }
            >
              6
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#212121",
                  "fontFamily": "Muli-SemiBold",
                  "fontSize": 17,
                }
              }
            >
               x 
            </Text>
            <View
              style={Object {}}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                üç∫
                 
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "color": "#212121",
                  "flex": 1,
                  "fontFamily": "Muli-SemiBold",
                  "fontSize": 17,
                }
              }
            >
              Beer
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#FFFFFF",
              "height": 1,
            }
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
